node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt

```
@@ -170,8 +170,8 @@ class GlobalTransientClassWhiteList(val delegate: ClassWhitelist) : MutableClass
 /**
  * A whitelist that can be customised via the [net.corda.core.node.SerializationWhitelist], since implements [MutableClassWhitelist].
  */
-class TransientClassWhiteList(val delegate: ClassWhitelist) : MutableClassWhitelist, ClassWhitelist by delegate {
-    val whitelist: MutableSet<String> = Collections.synchronizedSet(mutableSetOf())
+class TransientClassWhiteList(private val delegate: ClassWhitelist) : MutableClassWhitelist {
+    private val whitelist: MutableSet<String> = Collections.synchronizedSet(mutableSetOf())
 
     override fun hasListed(type: Class<*>): Boolean {
         return (type.name in whitelist) || delegate.hasListed(type)
```

I was thinking yesterday.... should we re-order this OR ?

Rationale: If we wanted to prevent the whitelisting by apps of certain types, with a blacklist, we need to check our delegate `ClassWhitelist` implementation first before the one they are allowed to add to.  Our blacklist implementation throws an exception rather than returning, so this would then work.

What do you think?  If we change it, can we write a test to make sure we don't re-order again later, and add a comment here that the order matters?

---

Very interesting trick, also agree.

---

Sure will make a change.